Non-unified build fixes August 2025

iangrunert opened the pull request:
#### c5a3f5b16fcb08b381ab1a343d82ae97ca7e9979
<pre>
Non-unified build fixes August 2025
<a href="https://bugs.webkit.org/show_bug.cgi?id=297747">https://bugs.webkit.org/show_bug.cgi?id=297747</a>

Reviewed by Tyler Wilcock.

Still doesn&apos;t fully build, currently fails on StyleValueTypes.h /
StyleGridTrackSize.cpp on an undefined template for
Blending&lt;GridTrackFitContentLength&gt;

Canonical link: <a href="https://commits.webkit.org/299270@main">https://commits.webkit.org/299270@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9b964603cb23665753fec64bf13fc10c1a9429ca

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117931 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37609 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28243 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124078 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69966 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119809 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38302 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46191 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89497 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59106 "Too many flaky failures: css3/filters/drop-shadow-current-color.html, fast/canvas/offscreen-no-script-context-crash.html, http/tests/resourceLoadStatistics/only-accept-first-party-cookies-with-third-party-cookie-blocking.html, imported/w3c/web-platform-tests/trusted-types/TrustedTypePolicyFactory-getAttributeType-event-handler-content-attributes.tentative.html, imported/w3c/web-platform-tests/trusted-types/set-event-handlers-content-attributes.tentative.html, storage/indexeddb/duplicates-private.html, storage/indexeddb/keypath-basics.html, storage/indexeddb/objectStore-required-arguments.html, webgl/2.0.0/conformance2/textures/canvas_sub_rectangle/tex-3d-r16f-red-float.html, webgl/2.0.0/conformance2/textures/canvas_sub_rectangle/tex-3d-rgb5_a1-rgba-unsigned_short_5_5_5_1.html, webgl/2.0.0/conformance2/textures/canvas_sub_rectangle/tex-3d-rgb8-rgb-unsigned_byte.html, webgl/2.0.0/conformance2/textures/image/tex-3d-rgb5_a1-rgba-unsigned_short_5_5_5_1.html, webgl/2.0.0/conformance2/textures/image_bitmap_from_blob/tex-2d-rgba8ui-rgba_integer-unsigned_byte.html, webgl/2.0.0/conformance2/textures/image_bitmap_from_canvas/tex-2d-rg8ui-rg_integer-unsigned_byte.html, webgl/2.0.0/conformance2/textures/image_bitmap_from_image_data/tex-2d-rgba8-rgba-unsigned_byte.html, webgl/2.0.0/conformance2/textures/image_bitmap_from_image_data/tex-2d-rgba8ui-rgba_integer-unsigned_byte.html, webgl/2.0.0/conformance2/textures/webgl_canvas/tex-2d-rgba16f-rgba-float.html, webgl/2.0.0/conformance2/textures/webgl_canvas/tex-3d-rgb565-rgb-unsigned_byte.html, webgl/2.0.y/conformance/glsl/bugs/conditional-discard-optimization.html, webgl/2.0.y/conformance/glsl/bugs/conditional-texture-fetch.html, webgl/2.0.y/conformance/glsl/misc/shader-with-non-reserved-words-5-of-8.html, webgl/2.0.y/conformance/ogles/GL/biuDepthRange/biuDepthRange_001_to_002.html, webgl/2.0.y/conformance/rendering/many-draw-calls.html, webgl/2.0.y/conformance/textures/image_bitmap_from_image/tex-2d-alpha-alpha-unsigned_byte.html (failure)") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120883 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30503 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105749 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69992 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29570 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23864 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67741 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/110042 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/99924 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24043 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127159 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/116441 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44834 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33774 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98168 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45195 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/101973 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/97956 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25000 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43347 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21328 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41268 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44706 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50380 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/145139 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44166 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37343 "Found 1 new JSC binary failure: testapi, Found 20045 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/array_sort2.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/blue_1086262.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/bug215238.shrua-2.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default, ChakraCore.yaml/ChakraCore/test/Function/FuncBody.bug227901.js.default ... (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47511 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45855 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e8599fcc096b8a285c67cac494afbe2aae6328e6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49732

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/39b4eb24e68a0f85e9d788590f9392e9f8e08f79. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49732

twilco commented:
Really sorry for all the trouble on the accessibility front.

> Current error I'm hitting is due to AccessibilityObject.h having a forward declared version of class AXObjectCache, so can't see it's CanMakeWeakPtr:

We really do want this function to be inlined since it's called all over the place. But maybe to just get yourself unblocked, move it to `AccessibilityObject.cpp` for now, and I'll come by in a separate PR later and make an `AccessibilityObjectInlines.h`. Would that fix this one?

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c772eca719c489eb23d186f424f1fa26efe1ceb5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49732

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6ab0c5354b65ed62f0ee4ea7bd4a7cd395d29fc8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49732

twilco commented:
I did open this PR:

https://github.com/WebKit/WebKit/pull/49844

But I can wait til you land this one and rebase mine so you don't have to deal with a rebase on your end.

iangrunert committed:
Non-unified build fixes August 2025
https://bugs.webkit.org/show_bug.cgi?id=297747

Reviewed by Tyler Wilcock.

Still doesn't fully build, currently fails on StyleValueTypes.h /
StyleGridTrackSize.cpp on an undefined template for
Blending<GridTrackFitContentLength>

Canonical link: https://commits.webkit.org/299270@main

webkit-commit-queue commented:
Committed 299270@main (c5a3f5b16fcb): <https://commits.webkit.org/299270@main>

Reviewed commits have been landed. Closing PR #49732 and removing active labels.
